### PR TITLE
i can now push data from the form to the database and then it goes to…

### PR DESCRIPTION
**Pull request by gidspektor**
… the website. i also added if statements to my functions and tested them, they stop null values being pulled from my database

### CODE REVIEW

File: form.css [hunk 1]
```diff
@@ -0,0 +1,19 @@
+body {
+    text-align: center;
+    font-weight: bold;
+}
+
```
Comment (olivergate) [inline form.css:5]: line break

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+
+
```
Comment (olivergate) [inline functions.php]: line break

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+function push_data( $bio, $life_now, $goals, pdo $db) {
```
Comment (olivergate) [inline functions.php]: type hinting required

File: index.php [hunk 4]
```diff
@@ -1,6 +1,7 @@
 
 <?php require 'pull_data.php';?>
 
+
```
Comment (olivergate) [inline index.php]: line break

File: pull_data.php [hunk 5]
```diff
@@ -1,11 +1,9 @@
 <?php
 
-
```
Comment (olivergate) [inline pull_data.php]: lb

File: pull_data.php [hunk 6]
```diff
@@ -1,11 +1,9 @@
...
 $db = new PDO('mysql:host=127.0.0.1;dbname=portfolio', 'root');
```
Comment (olivergate) [inline pull_data.php:6]: LB

File: pull_data.php [hunk 7]
```diff
@@ -1,11 +1,9 @@
...
 
 $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE,PDO::FETCH_ASSOC);
 
-
```
Comment (olivergate) [inline pull_data.php:8]: Oka, I canny be bothered to right Line break everywhere. Please get rid off unnecessary line breaks.

File: functions.php (hunk 3 above)
Comment (olivergate) [inline functions.php]: NO DOCBLOC?

File: pull_data.php [hunk 8]
```diff
@@ -16,37 +14,55 @@ function about_output(pdo $db) :array
 $about_result=about_output($db);
 
 
+
 /*
- * takes the array $about_result and turns it into a single key value of bio
+ * takes the array $about_result and turns it into a single key value of bio and finds out if the value is null
  *
  * @$param array this is an array of fields from the about_me section of the db
```
Comment (olivergate) [inline pull_data.php]: parameter name required here

File: pull_data.php [hunk 9]
```diff
@@ -16,37 +14,55 @@ function about_output(pdo $db) :array
...
  *
- * @returns string values from the key bio
+ * @returns string values from the key bio or if value is null it returns no
```
Comment (olivergate) [inline pull_data.php]: You either return the Bio as a string or a Boolean for which your return hints are not giving.

File: pull_data.php [hunk 10]
```diff
@@ -16,37 +14,55 @@ function about_output(pdo $db) :array
...
+        return 'false';
+        }
+        else{
+            return $about_result['bio'];
+        }
 }
 
 /*
- * takes the array $about_result and turns it into a single key value of life_now
+ * takes the array $about_result and turns it into a single key value of life_now and finds out if the value is null
```
Comment (olivergate) [inline pull_data.php]: The descriptions are not very descriptive. Are yo sure you mean key value?

Comment (olivergate) [reply pull_data.php]: Same problems with the above function about return type hinting and description and I imagine fr the one below.

These functions are so similar that they should really be one function. But ya know, up to you.

File: push_data.php [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+
+push_data($bio_push,$life_now_push,$goals_push,$db);
+
+?>
+
+<html>
+    <a href="form.php"><h3>click here to return to form</h3></a>
+    <a href="index.php"> <h3>click here to go back to webpage</h3></a>
+</html>
```
Comment (olivergate) [inline push_data.php:19]: Function Push_data has no doc bloc

Comment (olivergate) [reply push_data.php:19]: Function about_output has no doc bloc.

Comment (olivergate) [reply push_data.php:19]: Take the functions out of the files which use them and then require them in to add seperations of concern and so that when you run the tests it doesn't run a whole page.

Comment (olivergate) [reply push_data.php:19]: What is this file, if it isn't on use it shouldn't be here. gitignore please.

Comment (olivergate) [reply push_data.php:19]: NO TESTS WORK

Comment (olivergate) [reply push_data.php:19]: You have duplicate mages in your project. Keep them all in the img folder.

Comment (olivergate) [reply push_data.php:19]: There are no failure tests in here or type matching tests. Please update.

File: form.php [hunk 12]
```diff
@@ -0,0 +1,19 @@
+<?php
+require 'pull_data.php';
+?>
+
+
+
+<h1>Enter new information</h1>
+<form action="push_data.php" method="post">
+    <h3>edit bio</h3>
+    <input type="text" name="bio" value="<?php echo $bio;?>">
```
Comment (olivergate) [inline form.php]: input tag not closed

File: pull_data.php [hunk 13]
```diff
@@ -1,11 +1,9 @@
 <?php
 
-
 $db = new PDO('mysql:host=127.0.0.1;dbname=portfolio', 'root');
 
 $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE,PDO::FETCH_ASSOC);
 
-
 function about_output(pdo $db) :array
 {
```
Comment (olivergate) [inline pull_data.php]: Take functions outside of the place in which they are used. Preferably in a new file with a similar filename eg filename_function.php.
